docs/node/workflows.md

```diff
@@ -156,6 +175,30 @@ signalsLink="/docs/go/signals"
 
 ## Large Event Histories
 
+### Why `ContinueAsNew` is needed
+
 import SharedContinueAsNew from '../shared/continue-as-new.md'
 
 <SharedContinueAsNew />
+
+### The `ContinueAsNew` API
+
+Use the `Context.continueAsNew` API to instruct the Node SDK to restart `main` with a new starting value and a new event history.
+
+```ts
```

Comment:
Let's only use snippets with snipsync please.

This example (taken from the tests) is unclear, let's use something like (draft, not tested):
```ts
import { Context, sleep } from '@temporalio/workflow';

async function main(iteration = 0): Promise<void> {
  if (iteration === 10) return;
  console.log('Running Workflow iteration:', iteration);
  await sleep(1000);
  await Context.continueAsNew<typeof main>(iteration + 1); // must match the arguments expected by `main`
  // Unreachable code, continueAsNew is like `process.exit` and will stop execution once called.
}
```

You might want to mention that you can `ContinueAsNew` from a signal and you can also `ContinueAsNew` to a different Workflow.

Reply:
adding this code sample in

we cant use snipsync until we have the home for these samples.

Reply:
Please mention that you can ContinueAsNew from a signal and you can also ContinueAsNew to a different Workflow.